common/src/main/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoader.java

```diff
@@ -306,16 +308,16 @@ private static AlgorithmParameterSpec getLegacySpecForKeyStoreKey(@NonNull final
      * @return a {@link AlgorithmParameterSpec} for the keystore key (that we'll use to wrap the secret key).
      */
     private static AlgorithmParameterSpec getSpecForKeyStoreKey(@NonNull final Context context, @NonNull final String alias) {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P) {
-            return getLegacySpecForKeyStoreKey(context, alias);
-        } else {
-            int purposes = KeyProperties.PURPOSE_WRAP_KEY  | KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT;
+        if (CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_NEW_KEY_GEN_SPEC_FOR_WRAP) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            int purposes =  KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT;
```

Comment:
You still seem to be passing `purposes` below at line 313. Is this expected? I'm not sure if the PR is achieving what the title says

Reply:
I meant the PURPOSE_WRAP_KEY will be removed as that is the one that is causing the problem on this specific device. 
Hence the PR title. Changelog is not update at all. If you meant the commits, I have 2 commits. 
1st for removing the wrap purpose and 2nd for putting my whole code behind a flight just in case it is causing any new unknown problems.

Reply:
On top of it, I am also planning to push 1 more commit. Which will keep the PURPOSE_WRAP_KEY  purpose as well and catches the SecureImportUnavailableException for these special devices. It would then try again to generate a keypair with just the other 2 purposes.

Reply:
Ok so purpose itself is fine but only specific purpose is the problem?